File: src/SOP8/Base/Base.Controller/AccountController.cs
using Microsoft.AspNetCore.Mvc;
using Base.Account.IBLL;
using Base.Account.IBLL.Request;
using Base.Account.IBLL.Response;
using Base.Account.IBLL.Interface;
using Microsoft.AspNetCore.Cors;
using Response;

namespace Base.Controller
{
    using Options;

    [EnableCors("UnEPolicy")]
    [Route("api/[controller]/[action]")]
    [ApiController]
    public class AccountController : ControllerBase
    {
        private IProcessManager m_processManager = null;
        private ILoginOption m_loginOption = null;
        private IPasswordPolicy m_passwordPolicy = null;
        private IUserCreator m_userCreator = null;

        public AccountController(IProcessManager processManager, ILoginOption loginOption, IPasswordPolicy passwordPolicy, IUserCreator userCreator)
        {
            m_processManager = processManager;
            m_loginOption = loginOption;
            m_passwordPolicy = passwordPolicy;
            m_userCreator = userCreator;
        }

        [HttpGet]
        public string LoginKey(long num)
        {
            return m_processManager.GetLoginKey(num);
        }

        [HttpPost]
        public IActionResult LoginKey2([FromBody] RequestLoginKey data)
        {
            bool isExternalLogin = m_loginOption.ExternalLoginUrl != null && m_loginOption.ExternalLoginUrl.Trim().Length > 0;
            ResponseLoginKey response = m_processManager.RequestLoginKey(data, isExternalLogin);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult Login([FromBody] RequestLogin data)
        {
            LoginResult response = m_processManager.Login(data, m_loginOption.ExternalLoginUrl, m_loginOption.AutoLogin);

            if (response.User != null)
                response.User.Options = m_loginOption;

            return Ok(response);
        }

        [HttpPost]
        public IActionResult CheckLoginSession([FromBody] CheckLoginSession data)
        {
            LoginResult result = m_processManager.CheckLoginSession(data.UserNo, data.SessionKey);

            if (result.User != null)
                result.User.Options = m_loginOption;

            return Ok(result);
        }

        [HttpPost]
        public IActionResult RequestOptions([FromBody] RequestOption data)
        {
            ResponseOption response = m_processManager.GetOption(data);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult SaveOptions([FromBody] SaveOption data)
        {
            MessageResult response = m_processManager.SaveOptions(data);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult RequestAllSite()
        {
            ResponseSite response = m_processManager.GetAllSites();
            return Ok(response);
        }

        [HttpPost]
        public IActionResult RequestUserList([FromBody] RequestUserList data)
        {
            ResponseUserList response = m_processManager.GetUserList(data);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult RequestGradeList()
        {
            ResponseGradeList response = m_processManager.GetGradeList();
            return Ok(response);
        }

        [HttpPost]
        public IActionResult UpdateUserInfo([FromBody] RequestUpdateUserInfo data)
        {
            MessageResult response = m_processManager.UpdateUserInfo(data);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult DeleteUser([FromBody] RequestDeleteUser data)
        {
            MessageResult response = m_processManager.DeleteUser(data);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult CreateUser([FromBody] RequestNewUser data)
        {
            ResponseAccountUser response = m_processManager.AddNewUser(data, m_userCreator, m_passwordPolicy);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult RequestPasswordPolicy()
        {
            ResponsePasswordPolicy response = new ResponsePasswordPolicy(true, "");

            response.AllowCharacters = m_passwordPolicy.AllowCharacters;
            response.MinimumLength = m_passwordPolicy.MinimumLength;
            response.MaximumLength = m_passwordPolicy.MaximumLength;
            response.NeedCharacter = m_passwordPolicy.NeedCharacter;
            response.NeedLowerCase = m_passwordPolicy.NeedLowerCase;
            response.NeedUpperCase = m_passwordPolicy.NeedUpperCase;
            response.NeedAlphabet = m_passwordPolicy.NeedAlphabet;
            response.NeedNumber = m_passwordPolicy.NeedNumber;
            response.NeedNumberOrCharacter = m_passwordPolicy.NeedNumberOrCharacter;

            return Ok(response);
        }

        [HttpPost]
        public IActionResult RequestChangePassword([FromBody] RequestChangePassword data)
        {
            MessageResult response = m_processManager.ChangePassword(data);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult RequestTemporaryPasswordWithSMS([FromBody] RequestTemporaryPasswordWithSMS data)
        {
            MessageResult response = m_processManager.MakeTemporaryPasswordWithSMS(m_userCreator, m_passwordPolicy, data.UserName, data.PhoneNumber);
            return Ok(response);
        }

        [HttpPost]
        public IActionResult RequestTemporaryPasswordWithEmail([FromBody] RequestTemporaryPasswordWithEmail data)
        {
            MessageResult response = m_processManager.MakeTemporaryPasswordWithEmail(m_userCreator, m_passwordPolicy, data.UserName, data.Email);
            return Ok(response);
        }
    }
}

File: src/SOP8/Egovframework/Base/base-controller/src/main/java/egovframework/base/api/account/AccountController.java
package egovframework.base.api.account;

//import java.util.Map;
//import java.util.HashMap;
//import java.util.List;

import egovframework.base.account.ibll.interface_.IPasswordPolicy;
import egovframework.base.account.ibll.interface_.IUserCreator;
import egovframework.base.account.ibll.request.*;
import egovframework.base.account.ibll.response.*;
import egovframework.base.response.MessageResult;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import egovframework.base.account.ibll.IProcessManager;
import egovframework.base.api.options.ILoginOption;

@RestController
@RequestMapping("/api/Account")
public class AccountController {
	private final IProcessManager processManager;
	private final ILoginOption loginOption;
	private final IUserCreator userCreator;
	private final IPasswordPolicy passwordPolicy;

	@Autowired
	public AccountController(IProcessManager processManager, ILoginOption loginOption, IPasswordPolicy passwordPolicy, IUserCreator userCreator)
	{
		this.processManager = processManager;
		this.loginOption = loginOption;
		this.userCreator = userCreator;
		this.passwordPolicy = passwordPolicy;
	}

	@GetMapping("/LoginKey")
	@ResponseBody
	public String LoginKey(long num) {
		return this.processManager.getLoginKey(num);
	}

	@PostMapping("/LoginKey2")
	@ResponseBody
	public ResponseLoginKey LoginKey2(@RequestBody RequestLoginKey data) {
		boolean isExternalLogin = this.loginOption.getExternalLoginUrl() != null && !this.loginOption.getExternalLoginUrl().trim().isEmpty();
		return this.processManager.requestLoginKey(data, isExternalLogin);
	}

	@PostMapping("/Login")
	@ResponseBody
	public LoginResult Login(@RequestBody RequestLogin data) {
		return this.processManager.login(data, this.loginOption.getExternalLoginUrl(), this.loginOption.getAutoLogin());
	}

	@PostMapping("/CheckLoginSession")
	@ResponseBody
	public LoginResult CheckLoginSession(@RequestBody CheckLoginSession data) {
		return this.processManager.checkLoginSession(data.getUserNo(), data.getSessionKey());
	}

	@PostMapping("/RequestOptions")
	@ResponseBody
	public ResponseOption RequestOptions(@RequestBody RequestOption data) {
		return this.processManager.getOption(data);
	}

	@PostMapping("/SaveOptions")
	@ResponseBody
	public MessageResult SaveOptions(@RequestBody SaveOption data) {
		return this.processManager.saveOptions(data);
	}

	@PostMapping("/RequestAllSite")
	@ResponseBody
	public ResponseSite RequestAllSite() {
		return this.processManager.getAllSites();
	}

	@PostMapping("/RequestUserList")
	@ResponseBody
	public ResponseUserList RequestUserList(@RequestBody RequestUserList data) {
		return this.processManager.getUserList(data);
	}

	@PostMapping("/RequestGradeList")
	@ResponseBody
	public ResponseGradeList RequestGradeList() {
		return this.processManager.getGradeList();
	}

	@PostMapping("/UpdateUserInfo")
	@ResponseBody
	public MessageResult UpdateUserInfo(@RequestBody RequestUpdateUserInfo data) {
		return this.processManager.updateUserInfo(data);
	}

	@PostMapping("/DeleteUser")
	@ResponseBody
	public MessageResult DeleteUser(@RequestBody RequestDeleteUser data) {
		return this.processManager.deleteUser(data);
	}

	@PostMapping("/CreateUser")
	@ResponseBody
	public ResponseAccountUser CreateUser(@RequestBody RequestNewUser data) {
		return this.processManager.addNewUser(data, this.userCreator, this.passwordPolicy);
	}

	@PostMapping("/RequestPasswordPolicy")
	@ResponseBody
	public ResponsePasswordPolicy RequestPasswordPolicy() {
		ResponsePasswordPolicy response = new ResponsePasswordPolicy(true, "");

		response.setAllowCharacters(this.passwordPolicy.getAllowCharacters());
		response.setMinimumLength(this.passwordPolicy.getMinimumLength());
		response.setMaximumLength(this.passwordPolicy.getMaximumLength());
		response.setNeedCharacter(this.passwordPolicy.isNeedCharacter());
		response.setNeedLowerCase(this.passwordPolicy.isNeedLowerCase());
		response.setNeedUpperCase(this.passwordPolicy.isNeedUpperCase());
		response.setNeedNumber(this.passwordPolicy.isNeedNumber());

		return response;
	}

	@PostMapping("/RequestChangePassword")
	@ResponseBody
	public MessageResult RequestChangePassword(@RequestBody RequestChangePassword data) {
		return this.processManager.changePassword(data);
	}

	@PostMapping("/RequestTemporaryPasswordWithSMS")
	@ResponseBody
	public MessageResult RequestTemporaryPasswordWithSMS(@RequestBody RequestTemporaryPasswordWithSMS data) {
		return this.processManager.makeTemporaryPasswordWithSMS(this.userCreator, this.passwordPolicy, data.getUserName(), data.getPhoneNumber());
	}

	@PostMapping("/RequestTemporaryPasswordWithEmail")
	@ResponseBody
	public MessageResult RequestTemporaryPasswordWithEmail(@RequestBody RequestTemporaryPasswordWithEmail data) {
		return this.processManager.makeTemporaryPasswordWithEmail(this.userCreator, this.passwordPolicy, data.getUserName(), data.getEmail());
	}
}
